src/pages/studyView/charts/barChart/BarChart.tsx

```diff
@@ -157,10 +173,52 @@ export default class BarChart extends React.Component<IBarChartProps, {}> implem
         return additionRatio * STUDY_VIEW_CONFIG.thresholds.barRatio;
     }
 
+    @autobind
+    private onMouseMove(event: React.MouseEvent<any>): void {
+        this.mousePosition.x = event.pageX;
+        this.mousePosition.y = event.pageY;        
+    }
+
+     /*
+     * Supplies the BarPlot with the event handlers needed to record when the mouse enters
+     * or leaves a bar on the plot.
+     */
+    private get barPlotEvents(){
+        const self = this;
+        return [{
+            target: "data",
+            eventHandlers: {
+                onMouseEnter: () => {
+                    return [{
+                        target: "data",
+                        mutation: (event: any) => {
+                            self.currentBarIndex = event.datum.eventKey;
+                            self.toolTipRange = {
+                                start: self.barData[self.currentBarIndex].dataBin.start,
+                                end: self.barData[self.currentBarIndex].dataBin.end,
+                                special: self.barData[self.currentBarIndex].dataBin.specialValue,
+                            }
+                            self.sampleCount = event.datum.y;                            
+                        }
+                    }];
+                },
+                onMouseLeave: () => {
+                    return [{
+                        target: "data",
+                        mutation: () => {
+                            self.toolTipRange = null;
+                            self.sampleCount = -1;
```

Comment:
Why do we need to update sampleCount?

Reply:
We don't. I left it in as a way to preserve data honesty for anyone trying to debug this. When the mouse leaves the bar, we don't know what `sampleCount` is, because there is no bar. I thought  setting `sampleCount` to a clearly invalid value was the best way to indicate this.

Ultimately, this isn't a big deal for me. If you want me to remove it, that's not a problem. I could also switch `-1` to a static constant to cut down on the magic numbers a bit. I'll defer to you on this one.

Reply:
I think it's better to keep all this stuff on tooltipModel to make it super clear what the context of the data is.  When mouse leaves, you just set the model to null.

Reply:
Ok, yes. That is a good idea that does both things.